packages/compass-e2e-tests/helpers/commands/navigate-to-collection-tab.ts

```diff
@@ -29,7 +29,33 @@ async function navigateToCollection(
   );
   await sidebarFilterInputElement.setValue(collectionName);
   const collectionElement = await browser.$(collectionSelector);
-  await collectionElement.waitForDisplayed();
+
+  // If we're not finding the collection in the sidebar immediately,
+  // click the refresh button on the sidebar. This mostly handles the case
+  // in which we've connected Compass in a before() hook, but the
+  // insert-test-data hooks in the beforeEach() hook have not run yet,
+  // but also makes the tests more resilient in general.
+  let foundCollection = false;
+  const refreshSidebarPromise = async () => {
```

Comment:
I agree that the insert script that ran before all the tests wasn't going to scale (it just grew out of a quick script that inserted a few numbers after all) and it makes sense to have test files bring their own data.

But I don't really like this kind of workaround in a command. I'd rather fix the test file or the hooks if hooks run after we cal this command. This can hide flaky behaviour, but it is also totally normal for a collection to not appear in the sidebar in 1 second (think datalake or anything on atlas).

ie. the difference between working around our own tests vs working around a browser or webdriver issue.